contracts/modules/cover/Cover.sol

```diff
@@ -457,6 +490,35 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon {
     _tokenController.burnFrom(msg.sender, price);
   }
 
+  /* ========== Active cover amount tracking ========== */
+
+  function updateGlobalActiveCoverAmountPerAsset(uint period, uint amountToCover, uint24 assetId) internal {
+
+    uint activeCoverAmount = getGlobalActiveCoverAmountForAsset(assetId);
+    uint32 currentBucket = SafeUintCast.toUint32(block.timestamp / BUCKET_SIZE);
+
+    globalActiveCoverAmountPerAsset[assetId] = uint96(activeCoverAmount + amountToCover);
+    lastGlobalBuckets[assetId] = currentBucket;
+    globalActiveCoverAmountBucketsPerAsset[assetId][(block.timestamp + period) / BUCKET_SIZE] = uint96(amountToCover);
```

Comment:
Assign (block.timestamp + period) / BUCKET_SIZE to expiryBucket and also use SafeUintCast